app/waf/src/controllers/adc.controller.ts

```diff
@@ -341,11 +355,27 @@ export class AdcController extends BaseController {
       tenantId: await this.tenantId,
     });
 
-    if (adc.type === 'HW' && !(await this.untrustAdc(adc))) {
-      throw new HttpErrors.UnprocessableEntity('Fail to untrust device');
+    if (adc.type === 'HW') {
```

Comment:
I think we can attempt to untrust HW or VE before delete it, right? If no trustedDeviceId, it can exit gracefully. And next setup can be deleting VE instance and ports from OpenStack. The last step is delete HW or VE db record. Correct?